docs/manual/src/futures.md

```diff
@@ -0,0 +1,138 @@
+# Async/Future support
+
+UniFFI supports exposing async rust functions over the FFI. It can convert a Rust `Future`/`async fn` to and from foreign native futures (`async`/`await` in Python/Swift, `suspend fun` in Kotlin etc.)
+
+Check out the [examples](https://github.com/mozilla/uniffi-rs/tree/main/examples/futures) or the more terse and thorough [fixtures](https://github.com/mozilla/uniffi-rs/tree/main/fixtures/futures).
+
+Note that currently async functions are only supported by proc-macros, but UDL would be fairly simple enhancement.
+
+## Example
+
+This is the shortest "async sleep()" I could come up with!
+```Rust
+use std::time::Duration;
+use async_std::future::{timeout, pending};
+
+/// Async function that says something after a certain time.
+#[uniffi::export]
+pub async fn say_after(ms: u64, who: String) -> String {
+    let never = pending::<()>();
+    timeout(Duration::from_millis(ms), never).await.unwrap_err();
+    format!("Hello, {who}!")
+}
+```
+
+This can be called by the following Python code:
+```python
+import asyncio
+from uniffi_example_futures import *
+
+async def main():
+    print(await say_after(20, 'Alice'))
+
+if __name__ == '__main__':
+    asyncio.run(main())
+```
+
+This code uses `asyncio` to drive the future to completion, while our exposed function is used with `await`.
+
+In Rust `Future` terminology, this means the foreign bindings supply the "executor" - think event-loop, or async runtime. In this example it's `asyncio`. There's no requirement for a Rust event loop.
+
+There are [some great rustdoc docs](https://docs.rs/uniffi_core/latest/uniffi_core/ffi/rustfuture/index.html) on the implementation that's well worth a read.
+
+See the [foreign-executor fixture](https://github.com/mozilla/uniffi-rs/tree/main/fixtures/foreign-executor) for more implementation details.
+
+## How it works
+
+As [described in the documentation](https://docs.rs/uniffi_core/latest/uniffi_core/ffi/rustfuture/index.html),
+UniFFI generates code which uses callbacks from Rust futures back into that foreign "executor" to drive them to completion.
+Fortunately, each of the bindings and Rust have consistent models, so the discussion below is Python, but it's almost exactly the same in Kotlin and Swift.
```

Comment:
```suggestion
Fortunately, each of the bindings and Rust have similar models, so the discussion below is Python, but it's almost exactly the same in Kotlin and Swift.
```

consistent with what? I think you meant that they are similar and so discussing one implementation translates to the others easily?